### DISCUSSION TITLE
Truncating parameters of a large neural network to a smaller neural network?

### DISCUSSION THREAD

**Discussion by jeohjee** [netket/netket#1501] [Q&A] [answered]
I am wondering whether it is possible to first obtain the ground state with a large neural network (say few layers of G-CNN) and then, for later purposes such as time evolution, to somehow transfer the information of the original G-CNN to a new NN of fewer parameters. I understand that the accuracy of the variational wave function would go down. But as long as the most important features of the ground state and lowest excitation states are captured by the smaller NN, small deviations from the optimal solution would not maybe matter?

The only idea I was able to come up with would be to first optimize a large NN with N1 hidden layers and then decrease the number of hidden layers from N1 to N2 (N1 > N2) by using the parameters of first N2 layers of the large network as an input values to optimize the new network with N2 hidden layers.  But probably my question does not make even any sense.

**Comment by gcarleo** [comment, marked as answer]
It makes sense, but the only way to do that in a clean way is to optimize the fidelity between the large, pre-trained model, and the target smaller model. Fidelity optimization can be done in several ways, see e.g. one of our latest preprint, Sinibaldi et al. 2023

**Comment by jeohjee** [comment]
Thanks for the reply Giuseppe. I appreciate all the help you have given me!

I tested netket fidelity optimizer. I tried to truncate G-CNN of 2 layers of 2 features to one layer of two features for 4x4 triangular  J1-J2 Heisenberg model (J2 = 0.15 J1). I was able to reach infidelity of ~0.05. I guess this is still too much deviation from the wavefunction of the larger NN though. And truncating G-CNN to even more simple and primitive NN would probably lead to  even large infidelity unfortunately (I also tested with symmetric RBM which yielded infidelity of ~0.1).

**Comment by gcarleo** [comment]
I think an infidelity of 0.05 is not necessarily too bad (and a bit inevitable, since you are compressing the model), but it really depends on the precision you are trying to achieve
